socket.c

```diff
@@ -321,13 +322,13 @@ void *telnet_connection_handler_thread(void *socket_desc)
 
 			// Lock FTL data structure, since it is likely that it will be changed here
 			// Requests should not be processed/answered when data is about to change
-			enable_thread_lock();
+			shm_write_lock();
```

Comment:
Shouldn't we use `shm_read_lock();` here? The comment may be outdated (or was actually never correct...).